src/index.md

```diff
@@ -58,6 +59,86 @@ const end = view(Inputs.date({label: "End", value: getDateXDaysAgo(1) }));
   }</div>
 </div>
 
+<div class="divider"></div>
+
+<h4>Spark Retrieval Result Codes</h4>
+<body>This section shows the Spark Retrieval Result Codes breakdown. You can adjust the date range. Records exist for the last 90 days.</body>
+
+```js
+const combine = (obj, target, keys) => {
+  obj[target] = keys.reduce((acc, key) => acc + (obj[key] || 0), 0)
+  for (const key of keys) {
+    delete obj[key]
+  }
+}
+const clone = obj => JSON.parse(JSON.stringify(obj))
+const tidy = clone(SparkRetrievalResultCodes).flatMap(({ day, rates }) => {
+  for (const [key, value] of Object.entries(rates)) {
+    rates[key] = Number(value)
+  }
+
+  combine(rates, 'HTTP_5xx', [
+    'LASSIE_502',
+    'HTTP_500',
+    'HTTP_502',
+    'ERROR_503',
+    'ERROR_522',
+    'ERROR_520',
+    'ERROR_500',
+    'BAD_GATEWAY',
+    'GATEWAY_TIMEOUT'
+  ])
+  combine(rates, 'HTTP_4xx', [
+    'ERROR_429',
+    'ERROR_404',
+    'ERROR_403',
+    'ERROR_408'
+  ])
+  combine(rates, 'IPNI_NOT_ADVERTISED', [
```

Comment:
It's not IPNI that is not advertised, it's the retrieval.

```suggestion
  combine(rates, 'RETRIEVAL_NOT_ADVERTISED', [
```

Reply:
Is it the retrieval that's advertised though? In my understanding, retrieval is what happens after the data has been advertised. I called it `IPNI_NOT_ADVERTISED` because the data was not advertised to IPNI. Maybe `DATA_NOT_ADVERTISED`? Downside: It doesn't mention IPNI. Or: `IPNI_NO_ADVERTISEMENT`? I'm going with that for now

Reply:
After typing this, I think `IPNI_NOT_ADVERTISED` also works

Reply:
I prefer `IPNI_NO_ADVERTISEMENT` most

Reply:
cool, that's what we have now